src/js/actions/tools.js

```diff
@@ -417,7 +417,13 @@ define(function (require, exports) {
      * @return {Promise} Resolves to tool change
      */
     var selectTool = function (nextTool) {
-        var toolStore = this.flux.store("tool");
+        var toolStore = this.flux.store("tool"),
+            currentTool = toolStore.getCurrentTool();
+
+        // If same tool is being selected, make this a no-op
```

Comment:
The super select tool in the mask mode relies on the select action to reset the behavior in different scenarios. Can you make this instead of being a no-op, only to not uninstall border policies